src/emqx_tracer.erl

```diff
@@ -24,11 +24,18 @@
 %% APIs
 -export([ trace/2
         , start_trace/3
+        , start_trace/4
         , lookup_traces/0
         , stop_trace/1
+        , stop_trace/2
         ]).
 
--type(trace_who() :: {clientid | topic, binary()}).
+-ifdef(TEST).
+-export([is_match/5]).
+-endif.
+
+-type(trace_who_with_context() :: {clientid | topic, emqx_types:clientid() | emqx_types:topic(), binary(), [atom()]}).
+-type(trace_who() :: {clientid | topic, emqx_types:clientid() | emqx_types:topic(), binary()}).
```

Comment:
maybe make this an `opaque()`
i.e. no other module should have to know this is a 3-tuple.